app/routers/questions.py

```diff
@@ -14,3 +14,24 @@ async def get_question(question_id: str):
         status_code=status.HTTP_404_NOT_FOUND,
         detail=f"Question {question_id} not found",
     )
+
+
+@router.get("/get_by_question_set/{question_set_id}")
+async def get_questions_by_question_set(
+    question_set_id: str, skip: int = None, limit: int = None
+):
+    pipeline = [{"$match": {"question_set_id": question_set_id}}]
+
+    if skip:
+        pipeline.append({"$skip": skip})
+
+    if limit:
+        pipeline.append({"$limit": limit})
+
+    if (questions := list(client.quiz.questions.aggregate(pipeline))) is not None:
+        return questions
+
+    raise HTTPException(
+        status_code=status.HTTP_404_NOT_FOUND,
+        detail=f"Questions is question set {question_set_id} not found",
```

Comment:
Questions might not be present also because the skip value might be set to some invalid. This covers all the cases IMO.

```suggestion
        detail=f"No questions not found",
```